### PR TITLE
RELATED: ONE-3065 Scatter plot - production ready

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/153/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1396/ : FAILURE in 3m 39s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/156/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1398/ : SUCCESS in 2m 04s

**Comment by gdgate** [comment]
Merge Failed.

This change or one of its cross-repo dependencies was unable to be automatically merged with the current state of its repository. Please rebase the change and upload a new patchset.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/161/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1406/ : SUCCESS in 1m 42s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/167/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1411/ : SUCCESS in 2m 02s

File: examples/src/components/ScatterPlotExample.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    'rgba(195, 49, 73, 1)',
+    'rgba(168, 194, 86, 1)',
+    'rgba(243, 217, 177, 1)',
+    'rgba(194, 153, 121, 1)'
+];
+
+export class ScatterPlotExample extends Component {
+    onLoadingChanged(...params) {
+        // eslint-disable-next-line no-console
+        return console.log('ScatterPlotExample onLoadingChanged', ...params);
```
Comment (pbenes) [inline examples/src/components/ScatterPlotExample.jsx]: is the return here needed?

File: src/components/visualizations/chart/chartOptionsBuilder.ts [hunk 2]
```diff
@@ -319,19 +324,30 @@ export function getHeatMapSeries(
...
+function isPrimaryMeasuresBucketEmpty(mdObject: VisualizationObject.IVisualizationObjectContent) {
+    const primaryMeasuresBucket = get(mdObject, 'buckets', [])
+        .find(bucket => bucket.localIdentifier === 'measures');
+
+    const primaryMeasuresBucketItems = get(primaryMeasuresBucket, 'items', []);
+    return isEmpty(primaryMeasuresBucketItems);
+}
+
+function isSecondaryMeasuresBucketEmpty(mdObject: VisualizationObject.IVisualizationObjectContent) {
+    const secondaryMeasuresBucket = get(mdObject, ['buckets'], [])
```
Comment (pbenes) [inline src/components/visualizations/chart/chartOptionsBuilder.ts]: `['buckets']` here vs. `'measures'` (no array) above, unify?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/175/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1417/ : SUCCESS in 4m 34s

**Comment by ivan-nejezchleb** [comment]
extended test - cafe

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/176/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/499/ : FAILURE in 12s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/592/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/179/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/501/ : UNSTABLE in 13m 59s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/180/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1419/ : SUCCESS in 2m 08s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/181/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1420/ : FAILURE in 57s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/593/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/182/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/502/ : SUCCESS in 15m 39s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/186/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1425/ : SUCCESS in 2m 30s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/187/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1426/ : SUCCESS in 9m 31s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/595/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/188/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/503/ : UNSTABLE in 14m 37s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/189/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1427/ : SUCCESS in 4m 44s

File: src/components/visualizations/chart/chartOptionsBuilder.ts [hunk 3]
```diff
@@ -76,6 +76,12 @@ const sortedByMeasureTypes = [
     VisualizationTypes.FUNNEL
 ];
 
+const dontSupportStackingTypes = [
```
Comment (tomassatinsky) [inline src/components/visualizations/chart/chartOptionsBuilder.ts]: `unsupportedStackingTypes` would be better as we use verbs just for functions and methods

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/191/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1431/ : FAILURE in 54s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/203/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1441/ : FAILURE in 1s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/204/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1442/ : SUCCESS in 4m 03s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/600/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/205/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/507/ : SUCCESS in 16m 50s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/206/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1443/ : SUCCESS in 5m 29s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/207/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1444/ : SUCCESS in 1m 58s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/208/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1445/ : SUCCESS in 3m 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/220/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1458/ : SUCCESS in 4m 06s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/33/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1459/ : SUCCESS in 1m 46s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/602/ : SUCCESS in 16m 46s

**Pull request merged**